api/src/logic.ts

```diff
@@ -0,0 +1,29 @@
+import { ForestType, SiteConditionRule } from "./database/schema/forestType";
+
+export function recommend_plant(user_siteConditions: SiteConditionRule[], model_foresTypes: ForestType[]) {
+    
+    var closestMatching!: ForestType;
+    var counter:number = 0;
+
+    // iterate and add to siteConditions
+    var siteConditions: SiteConditionRule[] = [];
```

Comment:
you shouldn't need to make a copy of the site conditions here, just use the input array directly